src/dbus.cpp

```diff
@@ -610,6 +519,94 @@ dbus_manager::~dbus_manager()
     deinit();
 }
 
+DBusHandlerResult dbus_manager::filter_signals(DBusConnection* conn, DBusMessage* msg, void* userData) {
+    auto& manager = *reinterpret_cast<dbus_manager*>(userData);
+
+    for(auto& sig : manager.m_signals) {
+        if(manager.m_dbus_ldr.message_is_signal(msg, sig.intf, sig.signal)){
+            auto sender = manager.m_dbus_ldr.message_get_sender(msg);
+            if((manager.*(sig.handler))(msg, sender)) 
+                return DBUS_HANDLER_RESULT_HANDLED;
+            else
+                return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
+        }
+    }
+    return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
+}
+
+bool dbus_manager::handle_properties_changed(DBusMessage* msg, const char* sender) {
+    std::string source;
+
+    metadata meta;
+    parse_mpris_properties(m_dbus_ldr, msg, source, meta);
```

Comment:
Properly behaving players may just send "PlaybackStatus" so this just clears out metadata.

Reply:
Okay, didn't know that. Should I just copy the old metadata and have `parse_mpris_properties` overwrite it?

Reply:
Maybe. Depends how it meshes with the m_active_player idea

Reply:
I think we should be fine if we clear the metadata completely if we change the active player.
Do you know a player which doesn't send full status updates, so I can test that situation?

Reply:
Most not named "Spotify" but i'm using Audacious and VLC.

Reply:
Not sure how I missed that, then, but yes, VLC breaks it. I'm almost finished with the message wrapper, I'll commit a fix after that one.